lib/tapioca/runtime/trackers/constant_definition.rb

```diff
@@ -25,16 +25,17 @@ class ConstantLocation < T::Struct
 
           key = tp.self
 
-          if tp.path == "(eval)"
+          path = tp.path
+          if File.exist?(path)
+            loc = build_constant_location(tp, caller_locations)
+          else
             caller_location = T.must(caller_locations)
-              .drop_while { |loc| loc.path == "(eval)" }
+              .select { |loc| loc.path && File.exist?(loc.path) }
               .first
```

Comment:
Maybe `find` instead:
```
            caller_location = T.must(caller_locations)
              .find { |loc| loc.path && File.exist?(loc.path) }
```
Now that I look at this code, I have no idea why I didn't use `find` myself 😂